### PR TITLE
Rate limit calls

### PR DESCRIPTION
**Pull request by Velenir**
Sample implementation of rate limiting provider calls

### CODE REVIEW

**Comment by anxolin** [comment]
This PR is not super required right now, if we want to reuse the work, we could just disable by default the rate limit (by default is 0ms, that wouldn't add the middleware)

I understand this would allow us to have equivalent code, and be able to enable the rate limit for testing with just an env var

**Review by anxolin — COMMENTED** [review]
This PR is not super required right now, if we want to reuse the work, we could just disable by default the rate limit (by default is 0ms, that wouldn't add the middleware)

I understand this would allow us to have equivalent code, and be able to enable the rate limit for testing with just an env var

**Comment by Velenir** [comment]
Though `0` would remove the delay, it would still can end up >0 because of event queue
I vote for closing this PR, and reopening when/if needed

**Comment by gnosis-info** [comment]
Travis automatic deployment:
   * **🎩 [Dapp](https://pr1610--dexreact.review.gnosisdev.com)**: Testing web app
   * **📚 [Storybook](https://pr1610--dexreact.review.gnosisdev.com/storybook/)**: Component stories

File: src/api/wallet/composeProvider.ts [hunk 1]
```diff
@@ -276,6 +276,61 @@ export const composeProvider = <T extends Provider>(
...
+      queue.push(args)
+      if (!timer) {
+        processQueue() // start immediately on the first invocation
+        timer = setInterval(processQueue, delay)
+      }
+    }
+  }
+
+  const passThroughMware: JsonRpcMiddleware = (_req, _res, next) => next()
+  const rateLimitedPassThrough = rateLimit(passThroughMware, 100) // a more visible delay of 100ms
```
Comment (anxolin) [inline src/api/wallet/composeProvider.ts:322]: should we have an env for this?, and when the env is 0 we just don't add the middleware?

Comment (anxolin) [reply src/api/wallet/composeProvider.ts:322]: 0 should be the default of that env imo

File: src/api/wallet/composeProvider.ts [hunk 2]
```diff
@@ -276,6 +276,61 @@ export const composeProvider = <T extends Provider>(
     ),
   )
 
+  // Simple delay implementation
```
Comment (anxolin) [inline src/api/wallet/composeProvider.ts:279]: remove dead code

File: src/api/wallet/composeProvider.ts (hunk 1 above)
Comment (Velenir) [reply src/api/wallet/composeProvider.ts:322]: ok, that makes sense
let me do it quickly

**Review by anxolin — APPROVED** [review]

**Review by W3stside — APPROVED** [review]
the code looks fine... but I'm scared this will have some unintended side effects. Are we sure this won't unexpectedly drop transactions?

**Comment by Velenir** [comment]
> the code looks fine... but I'm scared this will have some unintended side effects. Are we sure this won't unexpectedly drop transactions?

It wouldn't, and it's disabled for now because of 0 delay

**Comment by anxolin** [comment]
@W3stside merging this one here!

**Pull request merged**